src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -1327,7 +1355,46 @@ MasterAggregateExpression(Aggref *originalAggregate, AttrNumber *columnId)
 	const AttrNumber argumentId = 1; /* our aggregates have single arguments */
 
 	if (aggregateType == AGGREGATE_COUNT && originalAggregate->aggdistinct &&
-		CountDistinctErrorRate != DISABLE_DISTINCT_APPROXIMATION)
+		CountDistinctErrorRate == DISABLE_DISTINCT_APPROXIMATION &&
+		walkerContext->repartitionSubquery)
+	{
+		Aggref *aggregate = (Aggref *) copyObject(originalAggregate);
+		List *aggTargetEntryList = aggregate->args;
+		TargetEntry *distinctTargetEntry = linitial(aggTargetEntryList);
+		List *varList = pull_var_clause_default((Node *) distinctTargetEntry->expr);
+		ListCell *varCell = NULL;
+		List *uniqueVarList = NIL;
+		int startColumnCount = walkerContext->columnId;
+
+		foreach(varCell, varList)
+		{
```

Comment:
Instead of a bit complex logic below, do you think would something like following works here?

```C
foreach (varCell, varList)
{
     uniqueVarList = list_append_unique_ptr(uniqueVarList, var)
}

foreach (varCell, uniqueVarList)
{
   var->varattno = startColumnCount + columnIndex;
   columnIndex++;
}
```